src/Test/Html/Events.elm

```diff
@@ -0,0 +1,149 @@
+module Test.Html.Events
+    exposing
+        ( Event(..)
+        , simulate
+        )
+
+{-|
+
+This module allows you to simulate events on Html nodes, the Msg generated
+by the event is returned so you can test it
+
+@docs Event, simulate
+
+-}
+
+import ElmHtml.InternalTypes exposing (ElmHtml(NodeEntry))
+import Json.Decode exposing (decodeString)
+import Json.Encode exposing (bool, encode, object, string)
+import Native.HtmlAsJson
+import Test.Html.Query as Query
+import Test.Html.Query.Internal as QueryInternal
+
+
+{-| Event constructors to simulate events
+-}
+type Event
```

Comment:
Why is that? They are only used for simulating events, during tests
What could be changed there would be changing `Events` on the `Facts` type from `Maybe Json.Decode.Value` to a `Dict String { decoder: Json.Decode.Decoder msg }`, eliminating the need to the `eventDecoder` native function I've added.